libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp

```diff
@@ -74,41 +74,44 @@ void AP_InertialSensor_SITL::generate_accel()
 
     for (uint8_t j = 0; j < nsamples; j++) {
 
-        float xAccel = sitl->state.xAccel;
-        float yAccel = sitl->state.yAccel;
-        float zAccel = sitl->state.zAccel;
+        Vector3f accel = Vector3f(sitl->state.xAccel,
```

Comment:
Interestingly,  if you tried to use braces to initialise (
```suggestion
        Vector3f accel = Vector3f{sitl->state.xAccel,
```
)
you would get a compilation error here as you'd be trying to initialise a Vector3f from three doubles.  The error you'd get is a narrowing conversion - indicating you're losing precision, which you are here.

Reply:
> The error you'd get is a narrowing conversion - indicating you're losing precision, which you are here.

right, but I'm happy to lose that precision. We're providing the data as 32 bit float anyway